designs/2024-support-ts-config-files/README.md

```diff
@@ -0,0 +1,130 @@
+
+- There should not be extra overhead for JavaScript users. This means this change should not have a significant impact (if any at all) affecting users who use plain JavaScript config files.
+- The external tools that are used to parse the config files written in TypeScript should not create side effects. Specifically, it is imperative that these tools do not interfere with Node.js's native module resolution system by hooking into or altering the standard `import/require` mechanisms. This means tools like [`ts-node`](https://github.com/TypeStrong/ts-node) and [`tsx`](https://github.com/privatenumber/tsx) might not be suitable for this purpose.
+
+So far the tool that seems to be the most suitable for this purpose is [`jiti`](https://github.com/unjs/jiti). It does not introduce side effects and performs well, demonstrating its reliability. It also seems to be more battle-tested given some established frameworks such as [Nuxt](https://github.com/nuxt/nuxt) and [Tailwind CSS](https://github.com/tailwindlabs/tailwindcss) have been using it to load their configuration files.
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+The documentation for this feature will be added to the [ESLint User Guide](https://eslint.org/docs/user-guide/configuring) page. The documentation will explain how to use TypeScript configuration files and the differences between JavaScript and TypeScript configuration files.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think
+    about any opposing viewpoints that reviewers might bring up.
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+This change will most likely require at least one external tool as either a dependency or a peer dependency.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+This goal is to minimize the disruption to existing users. The primary focus is to ensure that the existing JavaScript configuration files continue to work as expected. The changes will only affect TypeScript users who are using TypeScript configuration files. The changes will not affect the existing JavaScript configuration files.
+
+## Alternatives
```

Comment:
It would be worth listing the several alternatives you considered in your [PR](https://github.com/eslint/eslint/pull/18134).

Reply:
I do mention them in the `README.md` just not as "alternatives" since the maintainers were very clear that they wanted to steer clear of `import/require` related side effects and both [`tsx`](https://github.com/privatenumber/tsx) and [`ts-node`](https://github.com/TypeStrong/ts-node) take over module imporing and cause issues. In this case they don't seem to be suitable because we are going to manually work with `require.cache` and sometimes they can cause issues with the cache to get out of sync.

Reply:
I would find it helpful to see these mentioned in the alternatives section. In the alternatives section, you can explicitly mention pros/cons and why you decided against each alternative. This makes it more clear that you considered different options, as opposed to leaving this section blank.

Reply:
Sure I can do that.